src/sql/parts/notebook/models/notebookConnection.ts

```diff
@@ -0,0 +1,94 @@
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import { localize } from 'vs/nls';
+
+import { IConnectionProfile } from 'sql/parts/connection/common/interfaces';
+
+export namespace constants {
+	export const hostPropName = 'host';
+	export const userPropName = 'user';
+	export const knoxPortPropName = 'knoxport';
+	export const clusterPropName = 'clustername';
+	export const passwordPropName = 'password';
+	export const defaultKnoxPort = '30443';
+}
+/**
+ * This is a temporary connection definition, with known properties for Knox gateway connections.
+ * Long term this should be refactored to an extension contribution
+ *
+ * @export
+ * @class NotebookConnection
+ */
+export class NotebookConnection {
```

Comment:
I have temporarily moved this logic into Azure Data Studio. Long term some / all will move back out to an extension